graph-service/analyze.py

```
@@ -0,0 +1,115 @@
+    
+    if not gds:
+        logger.error("Unable to connect to GDS after multiple attempts. Skipping this analytics run.")
+        return
+
+    try:
+        driver = GraphDatabase.driver(NEO4J_URI, auth=(NEO4J_USER, NEO4J_PASSWORD))
+        with driver.session() as session:
+            # Abort run early if there are no organization nodes yet
+            org_count_res = session.run("MATCH (o:Organization) RETURN count(o) AS cnt").single()
+            if org_count_res["cnt"] == 0:
+                logger.info("No Organization nodes present yet – skipping this analytics run.")
```

Ruff (0.11.9)

String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)